projects/wave-core-new/src/lib/backend/backend.model.ts

```diff
@@ -132,3 +132,13 @@ export interface ToDict<T> {
 export interface RegisterWorkflowResultDict {
     id: UUID;
 }
+
+export interface WorkflowDict {
+    type: 'Vector' | 'Raster';
+    operator: OperatorDict;
+}
+
+export interface OperatorDict {
+    type: string;
+    params: { [key: string]: any};
```

Comment:
Can we somehow ensure that this is serializable?
The _Dict_ types have the property to be JSON-compliant, hence use default types such that you can call JSON.stringify or JSON.parse for conversion to string.

We allow either string, number, object of (self), list of (self)… :thinking:

Reply:
I currently only use it as the return type of the getWorkflow method so it only has to be deserializable. I could rename it to lift the JSON-compliance assumption. I am not sure how to even make it JSON compliant, maybe you can give me a hint.

Reply:
We can only allow primitive types, arrays and dictionaries, so something like this:

```
type ParamTypes = string | number | Array<ParamTypes> | { [key: string]: ParamTypes };

interface Params {
  [key: string]: ParamTypes;
}
```

Reply:
8edf44feb598071214a05e6562791997f23a4de5